tracecat/auth/sandbox.py

```diff
@@ -35,38 +33,51 @@ def __init__(
         role: Role | None = None,
         secrets: list[str] | None = None,
         target: Literal["env", "context"] = "env",
+        environment: str = DEFAULT_SECRETS_ENVIRONMENT,
     ):
         self._role = role or ctx_role.get()
         self._secret_paths: list[str] = secrets or []
-        self._secret_objs: list[Secret] = []
+        self._secret_objs: Sequence[Secret] = []
         self._target = target
-        self._context = {}
-        self._encryption_key = os.getenv("TRACECAT__DB_ENCRYPTION_KEY")
-        if not self._encryption_key:
-            raise ValueError("TRACECAT__DB_ENCRYPTION_KEY is not set")
+        self._context: dict[str, Any] = {}
+        self._environment = environment
+        try:
+            self._encryption_key = os.environ["TRACECAT__DB_ENCRYPTION_KEY"]
+        except KeyError as e:
+            raise KeyError("TRACECAT__DB_ENCRYPTION_KEY is not set") from e
 
     def __enter__(self) -> Self:
         if self._secret_paths:
             self._secret_objs = asyncio.run(self._get_secrets())
             self._set_secrets()
         return self
 
-    def __exit__(self, exc_type, exc_value, traceback) -> None:
+    def __exit__(
+        self,
+        exc_type: type[BaseException],
+        exc_value: BaseException,
+        traceback: TracebackType,
+    ) -> None:
         self._unset_secrets()
         if exc_type is not None:
             logger.error("An error occurred", exc_info=(exc_type, exc_value, traceback))
```

Comment:
More information error with suggested action to follow up?

Reply:
Doesn't catch a specific error here, so can't really add anything constructive